src/main/java/tuteez/logic/parser/RemarkCommandParser.java

```diff
@@ -19,29 +27,109 @@ public class RemarkCommandParser implements Parser<RemarkCommand> {
+
+        Index personIndex = parsePersonIndex(argMultimap);
+
+        if (isAddRemarkCommand(argMultimap)) {
+            return createAddRemarkCommand(personIndex, argMultimap);
+        }
+
+        if (isDeleteRemarkCommand(argMultimap)) {
+            return createDeleteRemarkCommand(personIndex, argMultimap);
+        }
+
+        throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, MESSAGE_REMARK_MISSING_COMMAND_TYPE));
+    }
+
+    private void validateBasicCommandFormat(String args) throws ParseException {
+        if (args.trim().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, RemarkCommand.MESSAGE_USAGE));
+        }
+    }
+
+    private void validateOperationType(ArgumentMultimap argMultimap) throws ParseException {
+        boolean hasAddPrefix = argMultimap.getValue(PREFIX_ADD_REMARK).isPresent();
+        boolean hasDeletePrefix = argMultimap.getValue(PREFIX_DELETE_REMARK).isPresent();
+
+        if (hasAddPrefix && hasDeletePrefix) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    MESSAGE_REMARK_MULTIPLE_OPERATIONS));
+        }
+
+        if (!hasAddPrefix && !hasDeletePrefix) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    MESSAGE_REMARK_MISSING_COMMAND_TYPE));
+        }
+    }
+
+    private Index parsePersonIndex(ArgumentMultimap argMultimap) throws ParseException {
+        String preamble = argMultimap.getPreamble().trim();
+
+        if (preamble.isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, MESSAGE_MISSING_PERSON_INDEX));
+        }
+
+        Index index;
+
         try {
-            personIndex = ParserUtil.parseIndex(argMultimap.getPreamble());
+            index = ParserUtil.parseIndex(argMultimap.getPreamble());
         } catch (ParseException pe) {
-            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, RemarkCommand.MESSAGE_USAGE), pe);
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT,
+                    MESSAGE_INVALID_PERSON_INDEX_FORMAT));
         }
+        return index;
+    }
+
+    private AddRemarkCommand createAddRemarkCommand(Index personIndex, ArgumentMultimap argMultimap)
+            throws ParseException {
+
+        assert argMultimap.getValue(PREFIX_ADD_REMARK).isPresent();
 
-        if (argMultimap.getValue(PREFIX_ADD_REMARK).isPresent()) {
-            String remarkString = argMultimap.getValue(PREFIX_ADD_REMARK).get();
-            return new AddRemarkCommand(personIndex, new Remark(remarkString));
+        String remarkString = argMultimap.getValue(PREFIX_ADD_REMARK).get();
+
+        if (remarkString.trim().isEmpty()) {
+            throw new ParseException(String.format(MESSAGE_INVALID_COMMAND_FORMAT, MESSAGE_EMPTY_REMARK));
```

Comment:
Actually would it be more seamless to ignore empty remarks? Just a thought, your idea definitely works thought!!

Reply:
Great catch, I've changed the code to use the `parseRemark` method in `ParserUtil` instead, so that it follows a similar implementation to `AddCommandParser` :)